scene/3d/bone_attachment_3d.cpp

```diff
@@ -52,55 +60,321 @@ void BoneAttachment3D::_validate_property(PropertyInfo &property) const {
+	}
+}
+
+TypedArray<String> BoneAttachment3D::get_configuration_warnings() const {
+	TypedArray<String> warnings = Node3D::get_configuration_warnings();
+
+	if (use_external_skeleton) {
+		if (external_skeleton_node_cache.is_null()) {
+			warnings.append(TTR("External Skeleton3D node not set! Please set a path to an external Skeleton3D node."));
+		}
+	} else {
+		Skeleton3D *parent = Object::cast_to<Skeleton3D>(get_parent());
+		if (!parent) {
+			warnings.append(TTR("Parent node is not a Skeleton3D node! Please use an extenral Skeleton3D if you intend to use the BoneAttachment3D without it being a child of a Skeleton3D node."));
+		}
+	}
+
+	if (bone_idx == -1) {
+		warnings.append(TTR("BoneAttachment3D node is not bound to any bones! Please select a bone to attach this node."));
+	}
+
+	return warnings;
+}
+
+void BoneAttachment3D::_update_external_skeleton_cache() {
+	external_skeleton_node_cache = ObjectID();
+	if (has_node(external_skeleton_node)) {
+		Node *node = get_node(external_skeleton_node);
+		ERR_FAIL_COND_MSG(!node, "Cannot update external skeleton cache: Node cannot be found!");
+
+		// Make sure it's a skeleton3D
+		Skeleton3D *sk = Object::cast_to<Skeleton3D>(node);
+		ERR_FAIL_COND_MSG(!sk, "Cannot update external skeleton cache: Skeleton3D Nodepath does not point to a Skeleton3D node!");
+
+		external_skeleton_node_cache = node->get_instance_id();
+	} else {
+		if (external_skeleton_node.is_empty()) {
+			BoneAttachment3D *parent_attachment = Object::cast_to<BoneAttachment3D>(get_parent());
+			if (parent_attachment) {
+				parent_attachment->_update_external_skeleton_cache();
+				if (parent_attachment->has_node(parent_attachment->external_skeleton_node)) {
+					Node *node = parent_attachment->get_node(parent_attachment->external_skeleton_node);
+					ERR_FAIL_COND_MSG(!node, "Cannot update external skeleton cache: Parent's Skeleton3D node cannot be found!");
+
+					// Make sure it's a skeleton3D
+					Skeleton3D *sk = Object::cast_to<Skeleton3D>(node);
+					ERR_FAIL_COND_MSG(!sk, "Cannot update external skeleton cache: Parent Skeleton3D Nodepath does not point to a Skeleton3D node!");
+
+					external_skeleton_node_cache = node->get_instance_id();
+					external_skeleton_node = get_path_to(node);
+				}
+			}
+		}
+	}
+}
+
 void BoneAttachment3D::_check_bind() {
-	Skeleton3D *sk = Object::cast_to<Skeleton3D>(get_parent());
-	if (sk) {
-		int idx = sk->find_bone(bone_name);
-		if (idx != -1) {
-			sk->bind_child_node_to_bone(idx, this);
-			set_transform(sk->get_bone_global_pose(idx));
+	Skeleton3D *sk = _get_skeleton3d();
+
+	if (sk && !bound) {
+		if (bone_idx <= -1) {
+			bone_idx = sk->find_bone(bone_name);
+		}
+		if (bone_idx != -1) {
+			sk->call_deferred("connect", "bone_pose_changed", callable_mp(this, &BoneAttachment3D::on_bone_pose_update));
 			bound = true;
+			call_deferred("on_bone_pose_update", bone_idx);
```

Comment:
On `call_deferred` you can use `SNAME` too, to avoid creating the `StringName` at runtime.